clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -3859,13 +3862,25 @@ def note_protocol_method : Note<
 def warn_availability_fuchsia_unavailable_minor : Warning<
   "Fuchsia API Level prohibits specifying a minor or sub-minor version">,
   InGroup<Availability>;
+def warn_availability_hlsl_unavailable_strict: Warning<
+  "strict parameter is not supported in HLSL">,
+  InGroup<Availability>;
+def warn_availability_environment_only_in_hlsl: Warning<
+  "environment parameter is supported only in HLSL">,
+  InGroup<Availability>;
```

Comment:
It seems to me like these two should be errors not warnings, and maybe we can collapse them down. Something like:
```suggestion
def err_availability_unexpected_parameter: Error<
  "unexpected parameter %0 in availability attribute, not permitted in %select{HLSL|C/C++}1">;
```